BXMSDOC-7408 - Restructured TOC and content for Kogito operator

hmanwani-rh opened the pull request:
See JIRA: https://issues.redhat.com/browse/BXMSDOC-7408

Restructured TOC - https://docs.google.com/document/d/1uft8GCXU9Add1bRCRNXE4iKhEUOSRzOZ3ZKZKxn3CAk/edit#heading=h.bbs08a9meltb

@ricardozanini, @mwringe, @sutaakar, @radtriste 
In this PR, I have arranged the existing content as per our restructured TOC, consisting of content for the sections that are already available in Kogito and blank sections for which we need to write the content. Once this PR is approved, I am going to comment out all the blank sections, so that they do not appear in Kogito documentation, and when we start writing content for a section, we can start with uncommenting the section name. 
Please review and let me know your thoughts. 

Following is the doc preview of Kogito documentation
[Kogito Documentation](http://file.pnq.redhat.com/~hmanwani/BXMSDOC-7408-Kogito/#chap-kogito-deploying-on-openshift)

sutaakar commented:
I think that whole chapter 3 could be adjusted.
There is explicitly mentioned `OpenShift` (i.e. `Deploying Kogito services on OpenShift` and so on).
If we want to propagate Kogito as usable for any Kubernetes provider we can adjust the OpenShift mentions.
What do you think?

sutaakar commented:
Chapter 3.4 `Kogito Operator and CLI` looks very similar to chapter 3.1 `Kogito on OpenShift`
One of those should be removed.

sutaakar commented:
Otherwise looks ok, once new sections are written then existing parts need to be adjusted to keep the docs consistent.

ricardozanini commented:
@hmanwani-rh, I agree with @sutaakar comments. I have small additions to this doc:

> Deploying Kogito services on OpenShift

This section should be renamed to `Kogito Operator and CLI.` Otherwise, it will sound like we only support OpenShift. But if this change will happen later on, I'm ok with it for now.

> Travel agency tutorial for Kogito services on OpenShift

This section should under "Examples provided with Kogito Operator".

Dumb question: What's the difference between "Installing the Kogito Operator **in** OpenShift" and "Installing the Kogito Operator **on** Kubernetes"?

We can remove "Installing the Kogito Operator on OpenShift Kubernetes Engine" with a note to follow the same instructions as in "Installing on k8s".

> Deploying a custom Kogito service in a Kubernetes cluster using custom configurations

We are missing the chapter number.

> OpenShift deployment options with the Kogito Operator and CLI (existing section that we have in Kogito)

As I commented in the TOC (sorry for only bring this up now), we are handing only the build feature, which applies only to OpenShift. So, it makes sense to use build instead of deployment. In every other section, we address the act of deploying a service in the cluster as "deployment."

> 3.4.7.3. Deploying Kogito services on OpenShift using the Kogito CLI

This can be `Deploying Kogito services on OpenShift or Kubernetes using the Kogito CLI`

I didn't tackle the content for now, since we are rewriting most of it.

hmanwani-rh commented:
> I think that whole chapter 3 could be adjusted.
> There is explicitly mentioned `OpenShift` (i.e. `Deploying Kogito services on OpenShift` and so on).
> If we want to propagate Kogito as usable for any Kubernetes provider we can adjust the OpenShift mentions.
> What do you think?

@sutaakar Thanks for raising this point. For now, I've just arranged the existing content based on the restructured TOC. Once we start rewriting the content we can adjust the mentions of OpenShift and make the doc more consistent. How about keeping it as is for now and later we'll work on it along with the content? wdyt?

hmanwani-rh commented:
> Chapter 3.4 `Kogito Operator and CLI` looks very similar to chapter 3.1 `Kogito on OpenShift`
> One of those should be removed.

Done, I've removed the Chapter 3.4 `Kogito Operator and CLI`.

hmanwani-rh commented:
> @hmanwani-rh, I agree with @sutaakar comments. I have small additions to this doc:
> 
> > Deploying Kogito services on OpenShift
> 
> This section should be renamed to `Kogito Operator and CLI.` Otherwise, it will sound like we only support OpenShift. But if this change will happen later on, I'm ok with it for now.
> 
> > Travel agency tutorial for Kogito services on OpenShift
> 
> This section should under "Examples provided with Kogito Operator".
> 
> Dumb question: What's the difference between "Installing the Kogito Operator **in** OpenShift" and "Installing the Kogito Operator **on** Kubernetes"?
> 
> We can remove "Installing the Kogito Operator on OpenShift Kubernetes Engine" with a note to follow the same instructions as in "Installing on k8s".
> 
> > Deploying a custom Kogito service in a Kubernetes cluster using custom configurations
> 
> We are missing the chapter number.
> 
> > OpenShift deployment options with the Kogito Operator and CLI (existing section that we have in Kogito)
> 
> As I commented in the TOC (sorry for only bring this up now), we are handing only the build feature, which applies only to OpenShift. So, it makes sense to use build instead of deployment. In every other section, we address the act of deploying a service in the cluster as "deployment."
> 
> > 3.4.7.3. Deploying Kogito services on OpenShift using the Kogito CLI
> 
> This can be `Deploying Kogito services on OpenShift or Kubernetes using the Kogito CLI`
> 
> I didn't tackle the content for now, since we are rewriting most of it.

@ricardozanini Thank you for the review. 

**Deploying Kogito services on OpenShift**

_This section should be renamed to Kogito Operator and CLI. Otherwise, it will sound like we only support OpenShift. But if this change will happen later on, I'm ok with it for now._

 Since there are related context and Id, which also needs to be changed and that will impact the cross-references as well. So, how about keeping the name as-is for now and we can make the changes, once we start with content? wdyt?

**Travel agency tutorial for Kogito services on OpenShift**

_This section should under "Examples provided with Kogito Operator"._

Done, I've moved this section under "Examples provided with Kogito Operator".

_Dumb question: What's the difference between "Installing the Kogito Operator in OpenShift" and "Installing the Kogito Operator on Kubernetes"?_

Oops, Installing the Kogito Operator in OpenShift is a typo, thanks for catching this. I've changed it to Installing the Kogito Operator **on** OpenShift. 

We use IN to convey that something is contained or inside, on the other hand, ON is used when you do not enclose something. With respect to installation or deployment, we do that on software or on a machine. For example, install bla-bla on your machine, but not in your machine because we'll not be opening the machine and installing it (Must be sounding like a lame example) :) 

Does this make sense? 

_We can remove "Installing the Kogito Operator on OpenShift Kubernetes Engine" with a note to follow the same instructions as in "Installing on k8s"._

Should we remove the entire section or we just need to write a note in the section?

**Deploying a custom Kogito service in a Kubernetes cluster using custom configurations**

_We are missing the chapter number._

Fixed it

**OpenShift deployment options with the Kogito Operator and CLI (existing section that we have in Kogito)**

_As I commented in the TOC (sorry for only bring this up now), we are handing only the build feature, which applies only to OpenShift. So, it makes sense to use build instead of deployment. In every other section, we address the act of deploying a service in the cluster as "deployment."_

Done, fixed it.

**3.4.7.3. Deploying Kogito services on OpenShift using the Kogito CLI**

_This can be Deploying Kogito services on OpenShift or Kubernetes using the Kogito CLI_

Done, fixed it.

hmanwani-rh commented:
@sutaakar @ricardozanini, I've made the changes as per your suggestions. Please take a look one more time and share your feedback.

ricardozanini commented:
> Since there are related context and Id, which also needs to be changed and that will impact the cross-references as well. So, how about keeping the name as-is for now and we can make the changes, once we start with content? wdyt?

Sounds good!

> Does this make sense?

Many thanks for the detailed explanation! I think I made some mistakes here and there.. hahaha

> Should we remove the entire section or we just need to write a note in the section?

I think a note in that section would make more sense!

Many thanks for handling this, @hmanwani-rh :)

sutaakar reviewed (approved):
Looks good to me

radtriste reviewed (approved):
lgtm
